arch/x86/core/prep_c.c

```diff
@@ -44,6 +49,8 @@ FUNC_NORETURN void z_x86_prep_c(void *arg)
 			    true);
 #endif
 
+	x86_irq_init();
```

Comment:
I find it more aesthetically pleasing to ifdef this under CONFIG_X86_64 rather than having to define an empty weak function earlier in this file so that the 32-bit build doesn't break, that weak function confused me for a bit. I don't care that much though.